internal/criocli/check_layer_state.go

```diff
@@ -0,0 +1,64 @@
+package criocli
+
+import (
+	"fmt"
+	"os"
+
+	cstorage "github.com/containers/storage"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+var CheckLayerStateCommand = &cli.Command{
+	Name: "check-layer-state",
+	Usage: `Check if state marker matches to the state flie contents.
+If they don't match, perform repairing of the storage.`,
+	Action: func(c *cli.Context) error {
+		config, err := GetConfigFromContext(c)
+		if err != nil {
+			return fmt.Errorf("unable to load configuration: %w", err)
+		}
+
+		store, err := config.GetStore()
+		if err != nil {
+			return fmt.Errorf("unable to open storage: %w", err)
+		}
+		defer func() {
+			if _, err := store.Shutdown(true); err != nil {
+				logrus.Errorf("Unable to shutdown storage: %v", err)
+			}
+		}()
+
+		statemarker := c.Args().Get(0)
+		statefile := c.Args().Get(1)
+		if statefile == "" {
+			return fmt.Errorf("state file path is required")
+		}
+
+		curMarker, err := os.ReadFile(statefile)
+		if err == nil {
+			if string(curMarker) == statemarker {
+				// no need to repair
+				return nil
+			}
+			checkOptions := cstorage.CheckEverything()
+			report, err := store.Check(checkOptions)
+			if err != nil {
+				return err
+			}
+			options := cstorage.RepairOptions{
+				RemoveContainers: true,
+			}
+			if errs := store.Repair(report, &options); len(errs) > 0 {
+				for _, err := range errs {
+					logrus.Error(err)
+				}
+				return errs[0]
+			}
+		} else if !os.IsNotExist(err) {
+			return err
+		}
```

Comment:
What about a small refactor to drop the indentation level here?

Reply:
Fixed